app/src/main/java/com/canvearth/canvearth/server/FBPixelManager.java

```diff
@@ -206,21 +211,27 @@ public void writePixelAsync(PixelData pixelData, Color color, @Nullable Runnable
     }
 
     // Please prefer writePixelAsync, for performance.
-    public void writePixelSync(PixelData pixelData, Color color) {
+    // returns last update pixel data.
+    public PixelData writePixelSync(PixelData pixelData, Color color) {
+        final PixelData returnPixelData = new PixelData(0, 0, 0);
         try {
             CountDownLatch latchForFinish = new CountDownLatch(1);
-            writePixelAsync(pixelData, color, latchForFinish::countDown);
+            writePixelAsync(pixelData, color, (PixelData lastUpdatedPixelData) -> {
+                returnPixelData.copyFrom(lastUpdatedPixelData);
+                latchForFinish.countDown();
+            });
             latchForFinish.await();
         } catch (InterruptedException e) {
             Log.e(TAG, e.getMessage());
         }
+        return returnPixelData;
     }
 
-
-    private void updateParent(FBPixel childOriginPixel, FBPixel childNewPixel,
+    // returns last updated pixel
+    private PixelData updateParent(FBPixel childOriginPixel, FBPixel childNewPixel,
```

Comment:
Why does this need to return the last updated pixel?

Reply:
I used this in performance test, to see which number of update occurred.
Maybe client-side don't really have to use this. Then simply ignore this return value

Reply:
Ok